Merge in cisagov/skeleton-tf-module and update code for use in the COOL

jsf9k opened the pull request:
# <!--- Provide a general summary of your changes in the Title above -->

## 🗣 Description

In this pull request I merge in cisagov/skeleton-tf-module and update code for use in the COOL.  The latter mostly involves turning the repo into a full-fledged Terraform role.

## 💭 Motivation and Context

The `dmarc.cyber.dhs.gov` email domain is only verified in the COOL DNS account.  Therefore the dmarc-import application must be moved there.

## 🧪 Testing

I have used these changes to successfully deploy dmarc-import to the COOL DNS account.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

jsf9k committed:
First commit

jsf9k committed:
Remove molecule test from Travis CI config

jsf9k committed:
Rename repo to skeleton-generic

jsf9k committed:
Merge pull request #1 from cisagov/improvement/rename

Rename repo to skeleton-generic

jsf9k committed:
Remove author information from README

jsf9k committed:
Merge pull request #2 from cisagov/improvement/remove_author_from_readme

Remove author information from README

jsf9k committed:
Update the pyenv installation instructions

In particular, mention pyenv-installer.

jsf9k committed:
Merge pull request #3 from cisagov/improvement/update_pyenv_instructions

Update the pyenv installation instructions

jsf9k committed:
Add pre-commit hook for docker-compose linting

jsf9k committed:
Remove black and pyupgrade from requirements-test

They are not needed there.

Also run pre-commit autoupdate to pick up the latest version of hooks.

jsf9k committed:
Merge pull request #4 from cisagov/improvement/add_precommit_hook_for_docker_compose

Add pre-commit hook for docker-compose linting

jsf9k committed:
Add bandit pre-commit hook

Also ran pre-commit autoupdate, which picked up a newer version of the
pyupgrade hook.

jsf9k committed:
Merge pull request #5 from cisagov/improvement/add_bandit

Add bandit pre-commit hook

jsf9k committed:
Ignore W503 from flake8

This is a warning about line breaks before binary operators.  It no
longer agrees with PEP8.  See, for example, here:
https://github.com/ambv/black/issues/21.

Guido agrees here:
https://github.com/python/peps/commit/c59c4376ad233a62ca4b3a6060c81368bd21e85b.

jsf9k committed:
Merge pull request #6 from cisagov/improvement/disable_w503

Ignore W503 from flake8

jsf9k committed:
Add pydoc warnings to flake8 config

Also add flake8-docstrings as an additional dependency in the
.pre-commit-config.yml file.

jsf9k committed:
Merge pull request #7 from cisagov/improvement/enable_docstring

Add pydoc warnings to flake8 config

felddy committed:
Add a configuration file for Bandit, and configure pre-commit to use it.

felddy committed:
Improve english

felddy committed:
Merge pull request #8 from cisagov/improvement-bandit-config

Improvement bandit config

jsf9k committed:
Run pre-commit autoupdate

Also update the URL for black since it moved GitHub orgs from ambv to
python.

jsf9k committed:
Merge branch 'develop' into improvement-bandit-config

jsf9k committed:
Merge pull request #9 from cisagov/improvement-bandit-config

Run pre-commit autoupdate

jsf9k committed:
Make .bandit.yml pass yamllint

jsf9k committed:
Merge pull request #10 from cisagov/improvement/make_bandit_config_pass_yamllint

Make .bandit.yml pass yamllint

jsf9k committed:
Enable TravisCI caching for pip and pre-commit plugins

This should speed up TravisCI builds.

jsf9k committed:
Update pre-commit plugins to latest versions

Via pre-commit autoupdate.

jsf9k committed:
Merge pull request #11 from cisagov/improvement/use_travisci_caching

Use TravisCI caching to speed up builds

jsf9k committed:
Add changes from running pre-commit autoupdate

jsf9k committed:
Merge pull request #12 from cisagov/improvement/update_precommit

Add changes from running pre-commit autoupdate

jsf9k committed:
Remove the terraform-docs hook

terraform-docs is currently broken with respect to terraform 0.12.

jsf9k committed:
Merge pull request #13 from cisagov/bugfix/turn_off_terraform_docs_hook

Remove the terraform-docs hook

jsf9k committed:
Configure prettier to ignore JSON files

JSON files are already being linted by the pretty-format-json hook.

jsf9k committed:
Merge pull request #14 from cisagov/improvement/tell_prettier_to_ignore_json_files

Configure prettier to ignore JSON files

mcdonnnj committed:
Update two text blocks to use the console tag instead of the bash tag for formatting.

mcdonnnj committed:
Merge pull request #15 from cisagov/update_contributing_text_blocks

Update two text blocks to use the console tag instead of the bash tag…

jsf9k committed:
Update README and CONTRIBUTING files for this project

jsf9k committed:
Add TF files for very basic EC2 instance deployment

jsf9k committed:
Add AWS region and AZ as inputs

Also use the bastion AMI for now, until the cisagov/skeleton-packer
repo is ready.

jsf9k committed:
Add an example showing how to use the TF module

This example deploys the module to the default VPC.

jsf9k committed:
Add a description to each of the outputs

jsf9k committed:
Add terraform foo to .gitignore

jsf9k committed:
Improve language in README

jsf9k committed:
Update pre-commit hooks via pre-commit autoupdate

jsf9k committed:
Add config file for markdownlint

This config file turns off the "line too long" rule when within a code
block or a table.  Often there is no way to avoid going over 80 lines
in such cases.

jsf9k committed:
Rename LICENSE.md to LICENSE, make the other changes that requires

The license text is not actually in Markdown, and adding the .md
extension actually makes it render a little differently than intended.

jsf9k committed:
Gussy up the ol' README.md file

jsf9k committed:
Fix typo

jsf9k committed:
Remove trailing periods from variable descriptions

jsf9k committed:
Merge pull request #16 from cisagov/improvement/markdownlint_config

Add markdownlint config

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into develop

jsf9k committed:
Add code in TravisCI config to install terraform

jsf9k committed:
Rename repo

The old name was too long, so I abbreviated "terraform" to "tf".

jsf9k committed:
Merge pull request #1 from cisagov/bugfix/name_change

Rename repo

jsf9k committed:
Fix typo

jsf9k committed:
Merge pull request #2 from cisagov/bugfix/fix_typo

Fix a typo

jsf9k committed:
Remove unnecessary characters

When defining the outputs, there is no need to use quotes, braces, or
dollar signs when referencing variables.

jsf9k committed:
Merge pull request #3 from cisagov/improvement/remove_unnecessary_quotes_on_variables

Remove unnecessary characters

felddy committed:
Add caching for curl and go.
This is derived from the patterns our other projects use.

felddy committed:
Merge pull request #4 from cisagov/improvement/travis_env

Add caching for curl and go.

felddy committed:
Bump terraform version.

felddy committed:
Merge pull request #5 from cisagov/improvement/terraform_0.12.6

Bump terraform version.

felddy committed:
Bump terraform version.

felddy committed:
Merge pull request #6 from cisagov/improvement/bump_terraform_version

Bump terraform version.

jsf9k committed:
Add updates from running pre-commit autoupdate

jsf9k committed:
Merge pull request #17 from cisagov/improvement/updates_from_precommit_autoupdate

Add updates from running pre-commit autoupdate

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/upstream_updates

jsf9k committed:
Merge pull request #7 from cisagov/improvement/upstream_updates

Upstream updates

dav3r committed:
Add updates from running pre-commit autoupdate

dav3r committed:
Merge pull request #18 from cisagov/improvement/updates_from_precommit_autoupdate

Add updates from running pre-commit autoupdate

felddy committed:
Replace Travis-CI with GitHub actions.

felddy committed:
Fix action shield link

felddy committed:
Disable ansible-lint until they fix their setup.py
See: https://github.com/ansible/ansible-lint/issues/590

felddy committed:
Remove extra line.

felddy committed:
Merge pull request #20 from cisagov/improvement/actions

Replace Travis-CI with GitHub actions

mcdonnnj committed:
Re-enabled ansible-lint and ran pre-commit autoupgrade.

mcdonnnj committed:
Merge pull request #21 from cisagov/pre-commit_autoupdate_2019-11-02

Re-enable ansible-lint and autoupgrade.

felddy committed:
Add GitHub action caching of pre-commit hooks and pip packages.

felddy committed:
Merge branch 'develop' into improvement/action_cache

felddy committed:
Remove search for previous pre-commit caches.
These seem to be very large caches.  Restoring an old one and updating 
it results in a cache larger than the maximum allowed cache size: 200MB. 
 
"Cache size of 254757924 bytes is over the 200MB limit, not saving 
cache."
So if the config changes it is best to just take the cache-miss and 
start from scratch.

felddy committed:
Merge pull request #22 from cisagov/improvement/action_cache

Add GitHub action caching of pre-commit hooks and pip packages.

felddy committed:
Switch from Travis to GitHub Actions.

felddy committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvement/actions

felddy committed:
Ensure curl cache directory exists.

felddy committed:
Add tmate to debug tar issue.

/bin/tar: /home/runner/.cache/curl: Cannot open: No such file or 
directory

felddy committed:
Use early expansion of curl cache variable.
Fixes problem with a tilde directory being created instead of using 
$HOME.

felddy committed:
Playing with quotes.

felddy committed:
Make the tmate session conditional on secret definition.

felddy committed:
Remove verbose curl flag and properly quote variables.

felddy committed:
Remove conditional tmate on secret. 
This isn't implemented nicely yet.

felddy committed:
Re-enable tmate

felddy committed:
Add robust handling of go cache location.

felddy committed:
Add secret triggered tmate session

felddy committed:
Add missing context to conditional check.

felddy committed:
Change tmate toggle variable.

felddy committed:
Change trigger value to avoid all 'true' in log being masked to '***'

felddy committed:
Try to simplify trigger value.

felddy committed:
Merge pull request #9 from cisagov/improvement/actions

Improvement/actions

felddy committed:
Force pre-commit hooks to use python3. 
Fixes #23

felddy committed:
Autoupdate pre-commit hooks.

felddy committed:
Merge pull request #24 from cisagov/improvement/pre_commit_default_python

Set a default language for pre-commit hooks

felddy committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvement/upstream

felddy committed:
Merge pull request #10 from cisagov/improvement/upstream

Improvement/upstream

felddy committed:
Add codeowners file with team OIS maintainers.
See: 
https://help.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners

felddy committed:
Merge pull request #25 from cisagov/improvement/codeowners

Add codeowners file with team OIS maintainers.

felddy committed:
Update CODEOWNERS

felddy committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvement/codeowners

felddy committed:
Merge pull request #11 from cisagov/improvement/codeowners

Improvement/codeowners

mcdonnnj committed:
Add seed-isort-config and isort to pre-commit hooks.
Add isort configuration file.

mcdonnnj committed:
Merge pull request #28 from cisagov/improvements/pre-commit_add_isort

Add isort tool to pre-commit hooks.

jsf9k committed:
Make workflow run when a PR is opened, synchronized, or reopened

A user forked cisagov/scan-target-data and created a pull request, but
the required GitHub Action(s) did not run.  This is presumably because
the user does not have Actions enabled in his or her fork.  Ideally,
the required Action(s) would run in cisagov/scan-target-data when a PR
to merge changes back is created.  Based on my reading of this link,
adding the "pull_request" event type should make this happen:
https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#pull-request-events-for-forked-repositories

jsf9k committed:
Improve list formatting

jsf9k committed:
Merge pull request #29 from cisagov/improvement/better-handle-prs-from-forked-repos

Make workflow run when a PR is opened, synchronized, or reopened

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/upstream-changes

jsf9k committed:
Merge pull request #12 from cisagov/improvement/upstream-changes

Pull in upstream changes

mcdonnnj committed:
Backported changes to CONTRIBUTING.md from the development guide.

mcdonnnj committed:
Merge pull request #31 from cisagov/improvements/add_wsl_instructions

Cleanup Linux instructions and add WSL references to CONTRIBUTING.md

mcdonnnj committed:
Update Python version used to 3.8
Update actions/checkout to v2
Update formatting to match downstream children

mcdonnnj committed:
Merge pull request #32 from cisagov/improvements/update_python_version_and_checkout_action

Update Python and actions/checkout Versions

mcdonnnj committed:
Run pre-commit autoupdate.

mcdonnnj committed:
Flip cache order to mirror how it is done downstream.

mcdonnnj committed:
Merge pull request #33 from cisagov/improvement/update_pre-commit_hooks

Update pre-commit hooks (and flip Actions cache order in workflow).

mcdonnnj committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvements/pull_in_upstream_changes

mcdonnnj committed:
All references to '-r' for pip calls have been replaced with the more verbose '--requirement'.

mcdonnnj committed:
Merge pull request #34 from cisagov/improvements/use_verbose_pip_switches

Use Verbose pip Switches

mcdonnnj committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvements/pull_in_upstream_changes

mcdonnnj committed:
Force trigger GitHub Actions to recover after https://www.githubstatus.com/incidents/fxbbtd7mhz1c

mcdonnnj committed:
Merge pull request #13 from cisagov/improvements/pull_in_upstream_changes

Pull in updates from skeleton-generic

felddy committed:
Autoupdate pre-commit hooks.  Add mypy.

felddy committed:
Merge pull request #35 from cisagov/improvement/static_type_checking

Autoupdate pre-commit hooks.  Add mypy.

jsf9k committed:
Specify the go version

This is required to get around the fact that terraform-docs does not
install with the default version of go that is present.

jsf9k committed:
Go ahead and start using v2 of the setup-go action, even though it's only in beta right now

jsf9k committed:
Merge pull request #14 from cisagov/bugfix/fix-broken-terraform-docs-install

Fix broken terraform-docs install

jsf9k committed:
Add .mypy_cache to .gitignore

jsf9k committed:
Merge pull request #36 from cisagov/improvement/add-mypy-cache-to-gitignore

Add .mypy_cache to .gitignore

mcdonnnj committed:
Incorporate the Python version into keys for pip and pre-commit caches.
This should resolve the issue seen when the Python version changes before there
is an update to .pre-commit-config.yml which results in pre-commit pointing to
a non-existent Python installation.

mcdonnnj committed:
Merge pull request #37 from cisagov/add_python_version_to_cache_keys

Incorporate the Python version into keys for pip and pre-commit caches.

dav3r committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into update_from_skeleton

mcdonnnj committed:
Change the cache paths from hardcoded values in their appropriate blocks to environment variables declared before the job block.

dav3r committed:
Merge pull request #15 from cisagov/update_from_skeleton

Pull in fix for pre-commit caching from upstream

mcdonnnj committed:
Merge pull request #38 from cisagov/change_cache_paths_to_env_variables

Change Cache Paths to Environment Variables

mcdonnnj committed:
Add a rule for markdownlint to allow multiple headers with the same name as long
as they are not nested in the same heading group.

mcdonnnj committed:
Merge pull request #39 from cisagov/add_mdl_rule_for_MD024

Add markdownlint Configuration Option for MD024

mcdonnnj committed:
Fixed broken hook id for pre-commit-terraform:
  terraform_validate_no_variables was changed to terraform_validate in the
  following commit:
  https://github.com/antonbabenko/pre-commit-terraform/commit/35e0356188b64a4c5af9a4e7200d936e514cba71.
Ran pre-commit autoupdate.

mcdonnnj committed:
Merge pull request #40 from cisagov/improvements/update_pre-commit

Update pre-commit Hooks and Fix Broken Hook ID

mcdonnnj committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvements/upstream_updates

mcdonnnj committed:
Add default provider information for the module.
Add Terraform initialization to the GitHub Actions workflow.

mcdonnnj committed:
Switch where the Terraform initialization takes place on the GitHub Actions
instance.

mcdonnnj committed:
We need to initialize the module and the example.

mcdonnnj committed:
Use my fork of the pre-commit-terraform hooks to allow the terraform_validate
hook to work with our directory tree.

mcdonnnj committed:
Adjust GitHub Actions workflow to eliminate superfluous 'cd ../..'.

mcdonnnj committed:
More default provider information for the module out of main.tf and into the
providers.tf per excellent feedback.

mcdonnnj committed:
Adjust 'terraform init' commands in the GitHub Actions workflow to eliminate
changing directories.

mcdonnnj committed:
+Add caching of .terraform directory to preserve provisioners between runs.
+Add directive to find and initialize Terraform directories.
-Removed directive that searched for Terraform directories.

mcdonnnj committed:
Switch back from personal fork of pre-commit-terraform now that my PR has been
merged.

mcdonnnj committed:
Ran `pre-commit autoupdate` to get the latest version of the
`pre-commit-terraform` hook since the PR I submitted,
https://github.com/antonbabenko/pre-commit-terraform/pull/100,
was approved. This will fix issues with `skeleton-tf-module` related to multiple
directories with Terraform code.

jsf9k committed:
Merge remote-tracking branch 'skeleton-tf-module/develop' into develop

jsf9k committed:
Remove some unnecessary files

jsf9k committed:
Resolve pre-commit complaints

jsf9k committed:
Resolve one more pre-commit complaint

jsf9k committed:
Separate main.tf into two files

mcdonnnj committed:
Explicitly install pre-commit hooks as its own step so it's clearer when the
failure is with setup rather than hooks running.

jsf9k committed:
Remove DMARC domain setup

This code is no longer needed, since it is now done in
cisagov/cool-dns-cyber.dhs.gov.

jsf9k committed:
Comment out Cognito configuration

We will get this working without Cognito, then we will go back and add
it in.

Also cleanup the code a little, and create three ES instances instead
of just one.  This should give us some cross-AZ resiliency.

jsf9k committed:
Reorder variables

jsf9k committed:
Promote the rule set name to a full-fledged variable

mcdonnnj committed:
Rename action to accurately reflect that we are only linting, not building.
Adjust description for pre-commit hook setup to better convey what is being
done.

mcdonnnj committed:
Sort .gitignore entries.

mcdonnnj committed:
Merge pull request #41 from cisagov/bugfix/update_pre-commit_for_pre-commit-terraform_hook

Update pre-commit Hooks to Include Bugfix

mcdonnnj committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvements/upstream_updates

mcdonnnj committed:
Merge pull request #16 from cisagov/improvements/upstream_updates

Pull in Upstream Changes

jsf9k committed:
Remove unused outputs.tf file

jsf9k committed:
Clean up resource names

jsf9k committed:
Remove region declaration

This messes up the role assumption.

jsf9k committed:
Clean up queue resource names

jsf9k committed:
Use resource attributes instead of var values

This forces resources to be constructed in the proper order.

jsf9k committed:
Fix alphabetization

jsf9k committed:
Fix a few resources name references that have changed

jsf9k committed:
Update README to match with latest changes

jsf9k committed:
Remove unnecessary examples directory

dav3r commented on README.md:
```diff
@@ -1,10 +1,10 @@
-that build the AWS infrastrusture used for parsing DMARC aggregate
-reports.  This repository goes along with
+`dmarc-import-terraform` contains the Terraform configuration files to
+build the AWS infrastrusture used for parsing DMARC aggregate reports.
```
It's not your change, but `infrastructure` is misspelled here.

jsf9k replied:
See commit baf88ac.

dav3r commented on README.md:
```diff
@@ -13,30 +13,70 @@ Here is a [Cloudcraft.co](https://cloudcraft.co) diagram of the basic
 
+## Examples ##
+
+No examples.
```
Can you add an issue to create an example here?

jsf9k replied:
See #12.

dav3r commented on iam.tf:
```diff
@@ -112,13 +112,13 @@ data "aws_iam_policy_document" "cloudwatch_doc" {
 # IAM policy document that that allows the Lambda function to invoke
 # itself.  This will be applied to the role we are creating.
-data "aws_iam_policy_document" "lambda_doc" {
+data "aws_iam_policy_document" "lambda_lambda" {
```
It's lambdas all the way down! 😄

felddy replied:
`Lambda` `Lambda` `Lambda`  and `OMEGA MU`

https://www.youtube.com/watch?v=5sqc98KHT4c

dav3r reviewed (approved):
So fresh, clean, and modern - strong work! 💪

jsf9k committed:
Turn on zone awareness

This allows the three nodes to be distributed across AZs.

mcdonnnj committed:
Merge branch 'develop' of github.com:cisagov/skeleton-tf-module into improvement/merge-in-skeleton-tf-module

jsf9k committed:
Fix typo

jsf9k committed:
Merge branch 'improvement/merge-in-skeleton-tf-module' of github.com:cisagov/dmarc-import-tf-module into improvement/merge-in-skeleton-tf-module

mcdonnnj reviewed (approved):
So fresh, so modern.

felddy reviewed (approved):
Looks great.  Not thrilled about this landing in the DNS account, but that is the world we live in.  Make sure you don't miss the video link: https://www.youtube.com/watch?v=5sqc98KHT4c

```
Clap your hands everybody, and everybody clap your hands!
We're Lambda Lambda Lambda and... Omega Mu
We come here on stage tonight to do a show for you
We've got a rockin rhythm and a high tech sound
That'll make you move your body down to the ground
```

mcdonnnj committed:
Switch the "aws_region" in providers.tf to use the variable defined in
variables.tf. Add default value and type to "aws_region" and
"aws_availability_zone" variables. Add type to "subnet_id" variable.

mcdonnnj committed:
Add "-upgrade=true" to Terraform initialization in the GitHub Actions workflow.

mcdonnnj committed:
Add note to README that mentions requiring a `terraform init` in any directory
containing Terraform code to pass `pre-commit`.

mcdonnnj committed:
Fix wording of Notes section in README per feedback.

mcdonnnj committed:
Merge pull request #17 from cisagov/improvements/cleanup_README_and_default_configuration

Fix Default Configuration and Add Note to README

jsf9k committed:
Merge remote-tracking branch 'skeleton-tf-module/develop' into improvement/merge-in-skeleton-tf-module

jsf9k committed:
Add ES domain as an output
